docs/source/extensions/logging.rst

```diff
@@ -75,55 +108,79 @@ Lightning offers automatic log functionalities for logging scalars, or manual lo
-
-* `on_epoch`: Automatically accumulates and logs at the end of the epoch. Defaults to True anywhere in validation or test loops, and in :func:`~~pytorch_lightning.core.lightning.LightningModule.training_epoch_end`.
-
-* `prog_bar`: Logs to the progress bar.
-
-* `logger`: Logs to the logger like Tensorboard, or any other custom logger passed to the :class:`~pytorch_lightning.trainer.trainer.Trainer`.
-
+The :meth:`~pytorch_lightning.core.lightning.LightningModule.log` method has a few options:
+
+* ``on_step``: Logs the metric at the current step.
+* ``on_epoch``: Automatically accumulates and logs at the end of the epoch.
+* ``prog_bar``: Logs to the progress bar (Default: ``False``).
+* ``logger``: Logs to the logger like ``Tensorboard``, or any other custom logger passed to the
+    :class:`~pytorch_lightning.trainer.trainer.Trainer` (Default: ``True``).
+
+.. list-table:: Default behavior of logging in Callback or LightningModule
+   :widths: 50 25 25
+   :header-rows: 1
+
+   * - Hook
+     - on_step
+     - on_epoch
+   * - on_before_backward, on_after_backward, on_before_optimizer_step, on_before_zero_grad
+     - False
+     - True
+   * - on_train_start, on_train_epoch_start, on_train_epoch_end, training_epoch_end
+     - False
+     - True
+   * - on_train_batch_start, on_train_batch_end
+     - False
+     - True
+   * - training_step, training_step_end
+     - True
+     - False
+   * - on_validation_start, on_validation_epoch_start, on_validation_epoch_end, validation_epoch_end
+     - False
+     - True
+   * - on_validation_batch_start, on_validation_batch_end, validation_step, validation_step_end
+     - False
+     - True
 
 .. note::
 
+    - The above config for ``val`` applies for ``test`` hooks as well.
+
     -   Setting ``on_epoch=True`` will cache all your logged values during the full training epoch and perform a
         reduction in ``on_train_epoch_end``. We recommend using `TorchMetrics <https://torchmetrics.readthedocs.io/>`_, when working with custom reduction.
 
     -   Setting both ``on_step=True`` and ``on_epoch=True`` will create two keys per metric you log with
-        suffix ``_step`` and ``_epoch``, respectively. You can refer to these keys e.g. in the `monitor`
+        suffix ``_step`` and ``_epoch`` respectively. You can refer to these keys e.g. in the `monitor`
         argument of :class:`~pytorch_lightning.callbacks.model_checkpoint.ModelCheckpoint` or in the graphs plotted to the logger of your choice.
 
 
-If your work requires to log in an unsupported function, please open an issue with a clear description of why it is blocking you.
+If your work requires to log in an unsupported method, please open an issue with a clear description of why it is blocking you.
 
 
-Manual logging
-==============
-If you want to log anything that is not a scalar, like histograms, text, images, etc... you may need to use the logger object directly.
+Manual logging Non-Scalar Artifacts
```

Comment:
I strongly suggest to either make all titles properly capitalized or not at all. For consistency. For the former, when in doubt, I recommend https://capitalizemytitle.com/

Reply:
thanks! updated!!